pkg/signup/service/signup_service.go

```diff
@@ -342,7 +363,7 @@ func (s *ServiceImpl) PhoneNumberAlreadyInUse(userID, phoneNumberOrHash string)
 		return errors.NewInternalError(err, "failed listing userSignups")
 	}
 	for _, signup := range userSignupList.Items {
-		if signup.Spec.Userid != userID && !states.Deactivated(&signup) { // nolint:gosec
+		if signup.Spec.Userid != userID && signup.Spec.Username != username && !states.Deactivated(&signup) { // nolint:gosec
```

Comment:
just out of curiosity - why do we need to verify also the `username`? wouldn't be the `userID` enough?

Reply:
What if the userID is empty though?  I think this is just safer to check both.